### PR TITLE
Fix xsave intrinsic test for clang, and gcc 8.2 or newer, and icc.

### PR DESCRIPTION
**Pull request by mtl1979**
* clang needs `<immintrin.h>` for `_xgetbv()` intrinsic function
* gcc 8 needs `<xsaveintrin.h>`
* gcc 9 and newer need `<immintrin.h>`
* icc needs `<immintrin.h>`

### CODE REVIEW

**Comment by phprus** [comment]
PR with ``internal_state`` ``ALIGNED_(16)``: #1663

**Comment by mtl1979** [comment]
> PR with `internal_state` `ALIGNED_(16)`: #1663

I have that user blocked for a good reason...

**Review by Dead2 — APPROVED** [review]

File: arch/x86/x86_features.c [hunk 1]
```diff
@@ -15,6 +15,13 @@
 #else
 // Newer versions of GCC and clang come with cpuid.h
 #  include <cpuid.h>
+#  ifdef X86_HAVE_XSAVE_INTRIN
+#    if __GNUC__ == 8
```
Comment (Dead2) [inline arch/x86/x86_features.c:19]: So this is only needed for 8.x, right? Not 9.x and newer or 7.x and older?

Also, this place you only check for 8.x, the other places you check that minor is 1 or more as well.

Intentional?

Comment (phprus) [reply arch/x86/x86_features.c:19]: ``X86_HAVE_XSAVE_INTRIN`` is not defined is gcc < 8

Comment (phprus) [inline arch/x86/x86_features.c:19]: Suggest:
```c
#    if defined(__GNUC__) && __GNUC__ == 8
```

And in other places.

Comment (mtl1979) [reply arch/x86/x86_features.c:19]: > ``X86_HAVE_XSAVE_INTRIN`` is not defined is gcc < 8

It's not defined for gcc < 8.2 as the implementation is buggy...

Comment (mtl1979) [reply arch/x86/x86_features.c:19]: > Suggest:
> ```c
> #    if defined(__GNUC__) && __GNUC__ == 8
> ```
> 
> And in other places.

It's not necessary as we already rely on anything undefined to be interpreted as `0`.

Comment (mtl1979) [reply arch/x86/x86_features.c:19]: @Dead2 AFAIK, Clang sets `__GNUC__` to `4`, Intel icc/icpc sets it to `6` or `7` depending on what operating system is used (`6` for Ubuntu 16.04, `7` for Ubuntu 18.04, essentially to same as default gcc/g++ version on the specific distribution)

**Pull request merged**